src/app/base/components/DhcpFormFields/MachineSelect/MachineSelect.tsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from "react";
+
+import { Label, useId } from "@canonical/react-components";
+import className from "classnames";
+import { useDispatch } from "react-redux";
+
+import SelectButton from "../../SelectButton";
+
+import MachineSelectBox from "./MachineSelectBox";
+
+import OutsideClickHandler from "app/base/components/OutsideClickHandler";
+import type { Machine } from "app/store/machine/types";
+import { useFetchMachine } from "app/store/machine/utils/hooks";
+import { actions as tagActions } from "app/store/tag";
+
+export enum Labels {
+  AppliesTo = "Applies to",
+  Loading = "Loading...",
+  ChooseMachine = "Choose machine",
+}
+
+type Props = {
+  label?: string;
+  onSelect: (machine: Machine | null) => void;
+  selected?: Machine["system_id"] | null;
+};
+
+export const MachineSelect = ({
+  label = Labels.AppliesTo,
+  onSelect,
+  selected = null,
+}: Props): JSX.Element => {
+  const dispatch = useDispatch();
+  const [isOpen, setIsOpen] = useState(false);
+  const selectId = useId();
+  const handleSelect = (machine: Machine | null) => {
+    setIsOpen(false);
+    onSelect(machine);
+  };
+
+  const { machine: selectedMachine } = useFetchMachine(selected, {
+    keepPreviousData: true,
+  });
+
+  useEffect(() => {
+    dispatch(tagActions.fetch());
+  }, [dispatch]);
+
+  return (
```

Comment:
Absolute positioning isn't related to whether we use a portal or not, it's more about the height of the element that can potentially exceed the height of the page and be clipped.

Agree it would be a good idea for the two to share a common base.

Reply:
Will follow up separately in https://github.com/canonical/maas-ui/issues/4397